### ISSUE TITLE
HDF5 Storing the `data` of higher level objects like PartcileContainer

### ISSUE THREAD

**Issue by itziakos** [simphony/simphony-common#15]
from @nathanfranklin:

How will we store the  ParticleContainer's data (or the `data` member of any high-level CUDS instances)?  Will we place those in a separate single-row table (e.g. at `/particle_container/foo/data`):

MaterialType | Label  
------------ | ------------- 
"MyType" | "my label"

Or can we just use the HDF5 attributes of the `/particle_container/foo/` Node to store the contents of the `data` DataContainer.  Here, I am unclear about the limitations of using HDF5-attributes.

Any thoughts on this part of the issue (storing ParticleContainer's `data`)?

**Comment by ahashibon** [comment]
The suggestion from @nathanfranklin , "to place those in a separate single-row table (e.g. at  /particle_container/foo/data )" sounds very reasonable, it does not require converting the data to a string and then parsing it (which may also make using SimPhoNy CUDS a bit awkward for third parties). The node attributes may still be used for example to store the name, or some short textual user description, or some metadata on the SimPhoNy version, etc... but not actual data related information. Does this make sense to you too?

**Comment by itziakos** [comment]
@ahashibon, I have provided an implementation proposal on this feature in https://publicwiki-01.fraunhofer.de/SimPhoNy-Project/index.php/Saving_the_%60data%60_information_of_a_CUDS_container

**Comment by mehdisadeghi** [comment]
Issues #83 and #84 have to be done first to close this issue.

**Comment by itziakos** [comment]
Issues #83 and #84 refer to the native python CUDS containers implementations not the HDF5 backed containers. So there is no dependency between them. Please also note that #83 and #84 are not part of this milestone.

Sorry for the confusion.

**Comment by mehdisadeghi** [comment]
The problem is that in current implementation, `H5CUDS` class has to copy values from native python particle containers to the h5 containers. Therefore problematic implementation of `data` property on the other classes will break the code. At least when I used `IndexedDataContainerTable` in my implementation it will not work unless python cuds objects provide proper `data` property of type `DataContainer`. That's why I did not implement this for Mesh and Lattice classes (in contradiction to the title of this issue, ”higher level objects“).

**Comment by itziakos** [comment]
Yup, you are right, #83 and #84 are blocking you from testing the fix for the hdf5 based containers. @roigcarlo, @SGGgarcia and @nathanfranklin, I think that we need to include #83 and #84 into the milestone, what do you think?

**Comment by nathanfranklin** [comment]
Yeah, that sounds good.

**Comment by SGGgarcia** [comment]
Yes, we should do that.

**Comment by roigcarlo** [comment]
Ok, I assign https://github.com/simphony/simphony-common/issues/83 to myself
